### PR TITLE
Add Transaction Model

### PR DESCRIPTION
**Pull request by rollingpencil**
Related to #80 

A transaction object consists of:
- id (UUID)
- Amount (Implemented with BigDecimal, follows [+/-][122][.{1,2}]: Must have +/-, fractional part optional but if needed has to be at most two decimal places
- Description (Non-null, at most 120 in length())
- Person (from AB3)
- Date (User input ddMMyyyy, display format is dd/MM/yyyy)

For now, `transaction.equal` method would return true if amount,description, person and date are the same, regardless  of UUID.

### CODE REVIEW

File: src/main/java/spleetwaise/transaction/model/transaction/Amount.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+ */
+public class Amount {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Amount should only contain digits up to 2 decimal points delimited by . and prefixed with +/-";
+
+    /*
+     * The first character of amount must be + or - and only allow precision up to 2 decimal places
+     */
+    public static final String VALIDATION_REGEX = "^(\\+|\\-)([\\d]+$|[\\d]+\\.[\\d]{1,2}$)";
```
Comment (SeeYangZhi) [inline src/main/java/spleetwaise/transaction/model/transaction/Amount.java:22]: Thinking if we want to allow the omission of a plus sign for ease of users. To be discussed.

**Review by SeeYangZhi — APPROVED** [review]
LGTM for v1.2. Will leave it for @seanlim to merge it.

File: src/main/java/spleetwaise/transaction/model/transaction/Amount.java (hunk 1 above)
Comment (seanlim) [reply src/main/java/spleetwaise/transaction/model/transaction/Amount.java:22]: Yeah I think leaving out the plus sign is a lot more intuitive.

Comment (seanlim) [reply src/main/java/spleetwaise/transaction/model/transaction/Amount.java:22]: Regex for this should be `^(\-)*([\d]+$|[\d]+\.[\d]{1,2}$)`

**Review by seanlim — APPROVED** [review]
LGTM! Good work!

**Pull request merged**